client/data/emails/use-add-email-forward-mutation.ts

```diff
@@ -0,0 +1,194 @@
+		queryClient.invalidateQueries( useGetDomainsQueryKey );
+	};
+
+	mutationOptions.onMutate = async ( variables ) => {
+		const { mailbox, destination } = variables;
+		suppliedOnMutate?.( variables );
+
+		await queryClient.cancelQueries( useGetEmailAccountsQueryKey );
+		await queryClient.cancelQueries( useGetDomainsQueryKey );
+
+		const previousEmailAccountsQueryData = queryClient.getQueryData< any >(
+			useGetEmailAccountsQueryKey
+		);
+		const emailForwards = previousEmailAccountsQueryData?.accounts?.[ 0 ]?.emails;
+
+		// Optimistically add email forward to `useGetEmailAccountsQuery` data
+		if ( emailForwards ) {
+			const newEmailForwards = orderBy(
+				[
+					...emailForwards,
+					{
+						domain: domainName,
+						email_type: 'email_forward',
+						is_verified: false,
+						mailbox,
+						role: 'standard',
+						target: destination,
+						temporary: true,
+						warnings: [],
+					},
+				],
+				[ 'mailbox' ],
+				[ 'asc' ]
+			);
+
+			queryClient.setQueryData( useGetEmailAccountsQueryKey, {
+				...previousEmailAccountsQueryData,
+				accounts: [
+					{
+						...previousEmailAccountsQueryData.accounts[ 0 ],
+						emails: newEmailForwards,
+					},
+				],
+			} );
+		}
+
+		const previousDomainsQueryData = queryClient.getQueryData< any >( useGetDomainsQueryKey );
+
+		// Optimistically increment the email_forward counter in `useGetDomainsQuery` data
+		if ( previousDomainsQueryData ) {
+			const selectedDomainIndex = previousDomainsQueryData.domains.findIndex(
+				( { domain }: { domain: string } ) => domain === domainName
+			);
+
+			if ( selectedDomainIndex >= 0 ) {
+				const selectedDomain = previousDomainsQueryData.domains[ selectedDomainIndex ];
+				const newDomains = [ ...previousDomainsQueryData.domains ];
+
+				newDomains.splice( selectedDomainIndex, 1, {
+					...selectedDomain,
+					email_forwards_count: selectedDomain.email_forwards_count + 1,
+				} );
+
+				queryClient.setQueryData( useGetDomainsQueryKey, {
+					...previousDomainsQueryData,
+					domains: newDomains,
+				} );
+			}
+		}
+
+		return {
+			[ JSON.stringify( useGetEmailAccountsQueryKey ) ]: previousEmailAccountsQueryData,
+			[ JSON.stringify( useGetDomainsQueryKey ) ]: previousDomainsQueryData,
+		};
+	};
+
+	mutationOptions.onError = ( error, variables, context ) => {
+		suppliedOnError?.( error, variables, context );
+
+		if ( context ) {
+			queryClient.setQueryData(
+				useGetEmailAccountsQueryKey,
+				context[ JSON.stringify( useGetEmailAccountsQueryKey ) ]
+			);
+
+			queryClient.setQueryData(
+				useGetDomainsQueryKey,
+				context[ JSON.stringify( useGetDomainsQueryKey ) ]
+			);
+		}
+
+		let errorMessage = translate(
+			'Failed to add email forwarding record. ' +
+				'Please try again or ' +
+				'{{contactSupportLink}}contact support{{/contactSupportLink}}.',
+			{
+				components: {
+					contactSupportLink: createElement( 'a', { href: CALYPSO_CONTACT } ),
```

Comment:
I'm curious, what benefits do you get from building this anchor with a function instead of doing this?

```suggestion
					contactSupportLink: <a href={ CALYPSO_CONTACT } />,
```

Reply:
I tried this, but kept running into syntax errors. Turns out that's because the file doesn't use a `tsx` file extension 🙄